java/src/jmri/implementation/AbstractStringIO.java

```diff
@@ -115,6 +116,17 @@ public String getBeanType() {
         return Bundle.getMessage("BeanNameStringIO");
     }
 
+    /**
+     * {@inheritDoc} 
+     * 
+     * Do a string comparison.
+     */
+    @CheckReturnValue
+    @Override
+    public int compareSystemNameSuffix(@Nonnull String suffix1, @Nonnull String suffix2, @Nonnull NamedBean n) {
+        return suffix1.compareTo(suffix2);
```

Comment:
Why does the default implementation in NamedBean not work?

Reply:
The problem is in [AbstractNamedBean](https://github.com/JMRI/JMRI/blob/master/java/src/jmri/implementation/AbstractNamedBean.java).

The default implementation of `compareSystemNameSuffix()` uses an `AlphanumComparator` which results in that the suffix `1` is the same as the suffix `01`. This is correct for turnouts and sensors, but not for Memories and StringIOs.